formfactory/templatetags/formfactory_tags.py

```diff
@@ -1,21 +1,81 @@
 from django import template
+from django.core.urlresolvers import reverse, resolve
 from django.http import Http404
 
 from formfactory import models
 
 register = template.Library()
 
 
-@register.inclusion_tag(
-    "formfactory/inclusion_tags/form_detail.html", takes_context=True
-)
-def render_form(context, object_or_slug):
-    render_dict = dict()
-    if isinstance(object_or_slug, models.Form):
-        render_dict["object"] = object_or_slug
-    elif isinstance(object_or_slug, basestring):
+@register.tag()
+def render_form(parser, token):
+    """{% render_form <form_slug> %}"""
+    tokens = token.split_contents()
+    if len(tokens) != 2:
+        raise template.TemplateSyntaxError(
+            "{% render_form <form_slug>/<object> %}"
+        )
+
+    return RenderFormNode(variable=tokens[1])
+
+
+class RenderFormNode(template.Node):
+
+    def __init__(self, variable):
+        self.variable = template.Variable(variable)
+
+    def render(self, context):
         try:
-            render_dict["object"] = models.Form.objects.get(slug=object_or_slug)
-        except models.Form.DoesNotExist:
-            raise Http404("No FormFactory Form matches the given query.")
-    return render_dict
+            variable = self.variable.resolve(context)
+        except template.VariableDoesNotExist:
+            variable = self.variable.var
+        default_msg = "No FormFactory Form matches the given query. %s" % self.variable
+
+        # If the variable is a string type we attempt to find object based on
```

Comment:
Can I suggest that you remove all references to persons in the comments, e.g "we".
Apologies for nitpicking